server/config.go

```diff
@@ -1,15 +1,89 @@
 package main
 
-// Link represents a pattern to autolink
-type Link struct {
-	Pattern              string
-	Template             string
-	Scope                []string
-	DisableNonWordPrefix bool
-	DisableNonWordSuffix bool
-}
-
-// Configuration from config.json
-type Configuration struct {
-	Links []*Link
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/mlog"
+	"github.com/mattermost/mattermost-server/model"
+)
+
+// Config from config.json
+type Config struct {
+	EnableAdminCommand bool
+	Links              []Link
+}
+
+// OnConfigurationChange is invoked when configuration changes may have been made.
+func (p *Plugin) OnConfigurationChange() error {
+	var c Config
+	err := p.API.LoadPluginConfiguration(&c)
+	if err != nil {
+		return err
+	}
+
+	for i := range c.Links {
+		err = c.Links[i].Compile()
+		if err != nil {
+			mlog.Error(fmt.Sprintf("Error creating autolinker: %+v: %v", c.Links[i], err))
+		}
+	}
+
+	p.updateConfig(func(conf *Config) {
+		*conf = c
+	})
+
+	go func() {
```

Comment:
The `go func() {}` here is a suggestion to work around https://github.com/mattermost/mattermost-server/pull/11780 which may not be available until mattermost-server 5.15 or later. Thanks @lieut-data for the suggestion.